gateway/handlers/reader.go

```diff
@@ -30,11 +30,14 @@ func MakeFunctionReader(metricsOptions metrics.MetricOptions, c *client.Client)
 
 		services, err := c.ServiceList(context.Background(), options)
 		if err != nil {
-			fmt.Println(err)
+			log.Println("Error getting service list:", err)
+			w.WriteHeader(http.StatusInternalServerError)
+			w.Write([]byte("Error getting service list"))
+			return
 		}
 
 		// TODO: Filter only "faas" functions (via metadata?)
-		var functions []requests.Function
+		functions := []requests.Function{}
```

Comment:
This changed?

Reply:
It did! When functions was `nil` (there were no services in the swarm) then `null` was returned in the body of the HTTP response. Now if there are no services, an empty array `[]` is returned instead.

Reply:
OK makes sense.